auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -527,15 +587,21 @@ static class Identity implements ModifiableRealmIdentity {
         private final Charset hashCharset;
         private final Encoding hashEncoding;
         private final SecretKey secretKey;
+        private final PrivateKey privateKey;
+        private final PublicKey publicKey;
+        private final File mainIndex;
 
-        Identity(final String name, final Path path, final IdentityLock lock, final Charset hashCharset, final Encoding hashEncoding, Supplier<Provider[]> providers, final SecretKey secretKey) {
+        Identity(final String name, final Path path, final IdentityLock lock, final Charset hashCharset, final Encoding hashEncoding, Supplier<Provider[]> providers, final SecretKey secretKey, final PrivateKey privateKey, final PublicKey publicKey, final File mainIndex) {
             this.name = name;
             this.path = path;
             this.lock = lock;
             this.hashCharset = hashCharset;
             this.hashEncoding = hashEncoding;
             this.providers = providers;
             this.secretKey = secretKey;
+            this.privateKey = privateKey;
+            this.publicKey = publicKey;
+            this.mainIndex = mainIndex;
```

Comment:
We're going to need to be a bit careful with the main index file because it's shared by the whole realm so we'll need to make sure that access to this file is synchronized.